applications/ColossalChat/coati/dataset/tokenization_utils.py

```diff
@@ -148,11 +152,14 @@ def tokenize_prompt(
     template = deepcopy(conversation_template)
     template.messages = []
 
+    if messages[0]["from"] == "system":
+        template.system_message = str(messages[0]["content"])
+        messages.pop(0)
```

Comment:
What will have the higher priority? 

What if I provide the system prompt in the config as well? 

In addition, if we provide `system` key in conversation and use `apply_chat_template`, it will automatically update using the system prompt.

Reply:
- The system prompt provided in messages has higher priority.
- if the first message is not from system. The system message in the config will apply. Otherwise will use the system message provided in the data sample
-  It depends on how you design your chat template, a very view of models just have their system message fixed. Most model's allow dynamically defines system message, here is an example from llama. 
"chat_template": "{% if messages[0]['role'] == 'system' %}{% set loop_messages = messages[1:] %}{% set system_message = messages[0]['content'] %}{% else %}{% set loop_messages = messages %}{% set system_message = false %}{% endif %}{% for message in loop_messages %}{% if (message['role'] == 'user') != (loop.index0 % 2 == 0) %}{{ raise_exception('Conversation roles must alternate user/assistant/user/assistant/...') }}{% endif %}{% if loop.index0 == 0 and system_message != false %}{% set content = '<<SYS>>\\n' + system_message + '\\n<</SYS>>\\n\\n' + message['content'] %}{% else %}{% set content = message['content'] %}{% endif %}{% if message['role'] == 'user' %}{{ bos_token + '[INST] ' + content.strip() + ' [/INST]' }}{% elif message['role'] == 'assistant' %}{{ ' '  + content.strip() + ' ' + eos_token }}{% endif %}{% endfor %}",